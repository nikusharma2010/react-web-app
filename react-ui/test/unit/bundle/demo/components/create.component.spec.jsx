import React from "react";
import { Provider } from 'react-redux';
import { create } from "react-test-renderer";
import { mount } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import CreateComponent from "../../../../../src/bundle/demo/components/create.component";
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({
    userState: {
        user: {},
        isLoading: false,
        error: 'Test'
    }
});
Enzyme.configure({ adapter: new Adapter() });

describe("Create Component", () => {
    test("it matches the snapshot", () => {
        const component = create(<Provider store={store}><CreateComponent /></Provider>);
        expect(component.toJSON()).toMatchSnapshot();
    });
});
describe("Form Submit", () => {
    let confirmation = false;
    let mockForm = {
        target:
        {
            employeeId: {
                value: 'test'

            },
            firstName: {
                value: 'test'

            },
            lastName: {
                value: 'test'

            },
            mobile: {
                value: 'test'

            },
            email: {
                value: 'test'

            }
        }
    };
    it('Check Value', () => {
        const wrapper = mount(<Provider store={store}><CreateComponent confirmation={confirmation} /></Provider>);
        wrapper.find('button').simulate('submit', mockForm);
        expect(wrapper.exists()).toEqual(true);
    });
});