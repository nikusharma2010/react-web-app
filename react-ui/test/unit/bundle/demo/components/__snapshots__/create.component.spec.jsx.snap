// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Component it matches the snapshot 1`] = `
<div
  className="styles__Row-sc-18xwbcb-2 MjgCm"
>
  <div
    className="styles__Row-sc-18xwbcb-2 MjgCm"
  >
    <div
      className="styles__Col-sc-18xwbcb-1 cOYgjR"
    >
      <a
        href="/users"
        onClick={[Function]}
      >
        <div
          className="styles__Button-sc-18xwbcb-5 styles__ButtonPrimary-sc-18xwbcb-6 YFSZa"
        >
          Back
        </div>
      </a>
    </div>
  </div>
  <div
    className="styles__FormDiv-sc-18xwbcb-10 hMsRmX"
  >
    <form
      onSubmit={[Function]}
    >
      <div>
        <h3>
          Create a New Employee record
        </h3>
      </div>
      <div
        className="styles__Row-sc-18xwbcb-2 MjgCm"
      >
        <div
          className="styles__Col-sc-18xwbcb-1 cOYgjR"
        >
          <label>
            Employee Number
          </label>
        </div>
        <div
          className="styles__Col-sc-18xwbcb-1 cOYgjR"
        >
          <input
            className="styles__Input-sc-18xwbcb-12 jqkTFa"
            id="employeeId"
            name="employeeId"
            type="text"
          />
        </div>
      </div>
      <div
        className="styles__Row-sc-18xwbcb-2 MjgCm"
      >
        <div
          className="styles__Col-sc-18xwbcb-1 cOYgjR"
        >
          <label>
            First Name
          </label>
        </div>
        <div
          className="styles__Col-sc-18xwbcb-1 cOYgjR"
        >
          <input
            className="styles__Input-sc-18xwbcb-12 jqkTFa"
            id="firstName"
            name="firstName"
            type="text"
          />
        </div>
      </div>
      <div
        className="styles__Row-sc-18xwbcb-2 MjgCm"
      >
        <div
          className="styles__Col-sc-18xwbcb-1 cOYgjR"
        >
          <label>
            Last Name
          </label>
        </div>
        <div
          className="styles__Col-sc-18xwbcb-1 cOYgjR"
        >
          <input
            className="styles__Input-sc-18xwbcb-12 jqkTFa"
            id="lastName"
            name="lastName"
            type="text"
          />
        </div>
      </div>
      <div
        className="styles__Row-sc-18xwbcb-2 MjgCm"
      >
        <div
          className="styles__Col-sc-18xwbcb-1 cOYgjR"
        >
          <label>
            Mobile
          </label>
        </div>
        <div
          className="styles__Col-sc-18xwbcb-1 cOYgjR"
        >
          <input
            className="styles__Input-sc-18xwbcb-12 jqkTFa"
            id="mobile"
            name="mobile"
            type="text"
          />
        </div>
      </div>
      <div
        className="styles__Row-sc-18xwbcb-2 MjgCm"
      >
        <div
          className="styles__Col-sc-18xwbcb-1 cOYgjR"
        >
          <label>
            Email
          </label>
        </div>
        <div
          className="styles__Col-sc-18xwbcb-1 cOYgjR"
        >
          <input
            className="styles__Input-sc-18xwbcb-12 jqkTFa"
            id="email"
            name="email"
            type="text"
          />
        </div>
      </div>
      <div
        className="styles__Row-sc-18xwbcb-2 MjgCm"
      >
        <div
          className="styles__Col-sc-18xwbcb-1 cOYgjR"
        >
          <button
            className="styles__SubmitButton-sc-18xwbcb-11 ftzQju"
            style={Object {}}
          >
            Add User
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
